mlir/lib/Interfaces/ViewLikeInterface.cpp

```diff
@@ -178,10 +188,34 @@ ParseResult mlir::parseDynamicIndexList(
       return failure();
     return success();
   };
+  auto parseColonType = [&]() -> ParseResult {
+    if (hasSameTypeDynamicValues) {
+      assert(valueTypes && "Expected non-null value types");
```

Comment:
Why is `valueTypes` requires when `hasSameTypeDynamicValues` is "true", but it is not required when `hasSameTypeDynamicValues` is "false"?